packages/react-components/src/components/trace-context-component.tsx

```diff
@@ -379,18 +386,44 @@ export class TraceContextComponent extends React.Component<TraceContextProps, Tr
                 text: `T1: ${t1} T2: ${t2} Delta: ${t2 - t1}`,
                 category: Messages.MessageCategory.TRACE_CONTEXT
             });
+
+            const payload = {
+                id: this.props.experiment.UUID,
+                timeRange: range,
+            } as TimeRangeUpdatePayload;
+            signalManager().fireSelectionRangeUpdated(payload);
+
             this.setState(prevState => ({
                 currentTimeSelection: new TimeRange(range.start, range.end, prevState.timeOffset)
             }), () => this.updateHistory());
         }
     }
 
     private handleViewRangeChange(viewRange: TimelineChart.TimeGraphRange) {
+
+        const payload = {
+            id: this.props.experiment.UUID,
+            timeRange: viewRange,
+        } as TimeRangeUpdatePayload;
+        signalManager().fireViewRangeUpdated(payload);
+
         this.setState(prevState => ({
             currentViewRange: new TimeRange(viewRange.start, viewRange.end, prevState.timeOffset)
         }), () => this.updateHistory());
     }
 
+    private emitTimeRangeData = (): void => {
```

Comment:
This is called in a loop, until indexing is finished. Does it need to be sent every iteration?